src/Catty/DataModel/UserDataContainer/UserDataContainer.swift

```diff
@@ -0,0 +1,233 @@
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+@objc(UserDataContainer)
+@objcMembers class UserDataContainer: NSObject {
+    private var variables: SynchronizedArray<UserVariable>
+    private var lists: SynchronizedArray<UserList>
+
+    override init() {
+        self.variables = SynchronizedArray<UserVariable>()
+        self.lists = SynchronizedArray<UserList>()
+    }
+
+    func allVariables() -> [UserVariable] {
+        var variables = [UserVariable]()
+        for index in 0..<self.variables.count {
+            if let variable = self.variables[index] {
+                variables.append(variable)
+            }
+        }
+        return variables
+    }
+
+    func allLists() -> [UserList] {
+        var lists = [UserList]()
+        for index in 0..<self.lists.count {
+            if let list = self.lists[index] {
+                lists.append(list)
+            }
+        }
+        return lists
+    }
+
+    func removeAllVariables() {
+        variables.removeAll()
+    }
+
+    func removeAllLists() {
+        lists.removeAll()
+    }
+
+    func getUserVariable(identifiedBy name: String) -> UserVariable? {
+        for index in 0..<variables.count {
+            if let variable = variables[index], variable.name == name {
+                return variable
+            }
+        }
+        return nil
+    }
+
+    func getUserList(identifiedBy name: String) -> UserList? {
+        for index in 0..<lists.count {
+            if let list = lists[index], list.name == name {
+                return list
+            }
+        }
+        return nil
+    }
+
+    func removeUserVariable(identifiedBy name: String) -> Bool {
+        if let _ = getUserVariable(identifiedBy: name) {
+            variables.remove(name: name)
+            return true
+        }
+        return false
+    }
+
+    func removeUserList(identifiedBy name: String) -> Bool {
+        if let _ = getUserList(identifiedBy: name) {
+            lists.remove(name: name)
+            return true
+        }
+        return false
+    }
+
+    @objc(containsVariable:)
+    func contains(_ variable: UserVariable) -> Bool {
+        variables.contains(variable)
+    }
+
+    @objc(containsList:)
+    func contains(_ list: UserList) -> Bool {
+        lists.contains(list)
+    }
+
+    @objc(addVariable:)
+    func add(_ variable: UserVariable) {
+        if !variables.contains(variable) {
+            variables.append(variable)
+        }
+    }
+
+    @objc(addList:)
+    func add(_ list: UserList) {
+        if !lists.contains(list) {
+            lists.append(list)
+        }
+    }
+
+    override func isEqual(_ object: Any?) -> Bool {
+        guard let userDataContainer = object as? UserDataContainer else { return false }
+
+        if self.allLists().count != userDataContainer.allLists().count || self.allVariables() != userDataContainer.allVariables() {
```

Comment:
Please revise this line. I think there is a `.count` missing for the variables. If tests are not failing, please add a test which makes this function fail before fixing it. Thanks!